carla/cf_models/catalog/dice/model.py

```diff
@@ -0,0 +1,87 @@
+import dice_ml
+import pandas as pd
+
+from ...api import CFModel
+
+
+class Dice(CFModel):
+    def __init__(self, mlmodel, data):
+        """
+        Constructor for Dice model
+        Implementation can be seen at https://github.com/interpretml/DiCE
+
+        Restrictions:
+        ------------
+        -   Only the model agnostic approach (backend: sklearn) is used in our implementation.
+        -   ML model needs to have a transformation pipeline for normalization, encoding and feature order.
+            See pipelining at carla/models/catalog/catalog.py for an example ML model class implementation
+
+        Parameters
+        ----------
+        mlmodel : models.api.MLModel
+            ML model to build counterfactuals for.
+        data : data.api.Data
+            Underlying dataset we want to build counterfactuals for.
+        """
+        # Prepare data for dice data structure
+        self.data = data
+        self._data_raw = data.raw.copy()
+        self._cont_feat = data.continous
+        self._target_name = data.target
+        self._dice_data = dice_ml.Data(
+            dataframe=self._data_raw,
+            continuous_features=self._cont_feat,
+            outcome_name=self._target_name,
```

Comment:
If we only use the data, cont_feat, target etc to pass into the dice model, there is no need to have them as object properties

Reply:
Right, makes it more simple. Will change it in the next commit.